chalk-solve/src/coherence.rs

```diff
@@ -71,20 +72,19 @@ impl<I: Interner> SpecializationPriorities<I> {
 #[derive(Copy, Clone, Default, PartialOrd, Ord, PartialEq, Eq, Debug)]
 pub struct SpecializationPriority(usize);
 
-impl<'db, I> CoherenceSolver<'db, I>
+impl<'db, I, S, SC> CoherenceSolver<'db, I, S, SC>
 where
     I: Interner,
+    S: Solver<I>,
```

Comment:
Can we possibly make this use a `dyn` trait? I think I'd prefer that to keep the generics under control.

Reply:
But I'd be happy to consider all that as follow-up work

Reply:
I'll file an issue for this.